components/omnibox/browser/brave_search_provider.cc

```diff
@@ -41,5 +41,9 @@ bool BraveSearchProvider::IsQueryPotentiallyPrivate() const {
     return true;
   }
 
+  if (!client()->IsQuerySafeToSearchSuggestions(input_text)) {
```

Comment:
I assume it's UI thread.
How much time it takes to perform a single call ?
Do we call that function for each omnibox change?

Reply:
0.2 ms looks good. Could also test it on the large requests (~50 chars)
Also would be nice to add UMA to easily estimate the perf, especially if we change that code one day.

Reply:
We don't do such suggestions for query more than 50 chars.
below is the result of long query almost 50 chars.
```
[1713:259:0827/203934.949709:ERROR:chrome_autocomplete_provider_client.cc(146)] IsQuerySafeToSearchSuggestions query: abcdefghijklmnadsfasdfasdfasasdfadsfadsf, time: 0.00017 s
[1713:259:0827/203935.130175:ERROR:chrome_autocomplete_provider_client.cc(146)] IsQuerySafeToSearchSuggestions query: abcdefghijklmnadsfasdfasdfasasdfadsfadsf, time: 0.000221 s
[1713:259:0827/203935.310075:ERROR:chrome_autocomplete_provider_client.cc(146)] IsQuerySafeToSearchSuggestions query: abcdefghijklmnadsfasdfasdfasasdfadsfadsf, time: 0.000193 s
[1713:259:0827/203935.613589:ERROR:chrome_autocomplete_provider_client.cc(146)] IsQuerySafeToSearchSuggestions query: abcdefghijklmnadsfasdfasdfasasdfadsfadsf a, time: 0.000231 s
[1713:259:0827/203935.760102:ERROR:chrome_autocomplete_provider_client.cc(146)] IsQuerySafeToSearchSuggestions query: abcdefghijklmnadsfasdfasdfasasdfadsfadsf af, time: 0.000263 s
[1713:259:0827/203935.815895:ERROR:chrome_autocomplete_provider_client.cc(146)] IsQuerySafeToSearchSuggestions query: abcdefghijklmnadsfasdfasdfasasdfadsfadsf afa, time: 0.000259 s
[1713:259:0827/203935.906059:ERROR:chrome_autocomplete_provider_client.cc(146)] IsQuerySafeToSearchSuggestions query: abcdefghijklmnadsfasdfasdfasasdfadsfadsf afad, time: 0.000279 s
[1713:259:0827/203935.931885:ERROR:chrome_autocomplete_provider_client.cc(146)] IsQuerySafeToSearchSuggestions query: abcdefghijklmnadsfasdfasdfasasdfadsfadsf afadf, time: 0.000221 s
[1713:259:0827/203936.018674:ERROR:chrome_autocomplete_provider_client.cc(146)] IsQuerySafeToSearchSuggestions query: abcdefghijklmnadsfasdfasdfasasdfadsfadsf afadfa, time: 0.000295 s
[1713:259:0827/203936.038725:ERROR:chrome_autocomplete_provider_client.cc(146)] IsQuerySafeToSearchSuggestions query: abcdefghijklmnadsfasdfasdfasasdfadsfadsf afadfad, time: 0.000261 s
[1713:259:0827/203936.075373:ERROR:chrome_autocomplete_provider_client.cc(146)] IsQuerySafeToSearchSuggestions query: abcdefghijklmnadsfasdfasdfasasdfadsfadsf afadfads, time: 0.000245 s
[1713:259:0827/203936.119709:ERROR:chrome_autocomplete_provider_client.cc(146)] IsQuerySafeToSearchSuggestions query: abcdefghijklmnadsfasdfasdfasasdfadsfadsf afadfadsf, time: 0.000251 s
```